lightning/src/ln/onion_payment.rs

```diff
@@ -473,27 +473,23 @@ where
 	NS::Target: NodeSigner,
 	L::Target: Logger,
 {
-	macro_rules! return_malformed_err {
-		($msg: expr, $err_code: expr) => {
-			{
-				log_info!(logger, "Failed to accept/forward incoming HTLC: {}", $msg);
-				let (sha256_of_onion, failure_code) = if msg.blinding_point.is_some() {
-					([0; 32], INVALID_ONION_BLINDING)
-				} else {
-					(Sha256::hash(&msg.onion_routing_packet.hop_data).to_byte_array(), $err_code)
-				};
-				return Err(HTLCFailureMsg::Malformed(msgs::UpdateFailMalformedHTLC {
-					channel_id: msg.channel_id,
-					htlc_id: msg.htlc_id,
-					sha256_of_onion,
-					failure_code,
-				}));
-			}
-		}
-	}
+	let encode_malformed_error = |message: &str, err_code: u16| {
```

Comment:
Small reviewability nit: separating the macro conversion in a commit would be slightly better I think.

Reply:
Yes, nicer. Thank you.